src/azul/service/hca_response_v5.py

```diff
@@ -412,6 +412,11 @@ def _make_contributor_matrices(cls, catalog, files):
         sorted_dimensions = sorted(distinct_values,
                                    key=lambda k: len(distinct_values[k]))
 
+        # Verify that every stratum uses the same dimensions
```

Comment:
Where does this requirement come from?

Reply:
This is a requirement for the construction of the tree due to the set depth of the NestedDict. If a strata doesn't have the same dimensions as the other strata, the node where the file URL would go will be a NestedDict instead of a List.

Reply:
That's not a requirement, but an implementation concern. It could be that the use of NestedDict or aspects of its implementation are inadequate to meet the requirements. Is there a an actual requirement for this?

Reply:
Created #2443 to allow stratification tree with varying dimensions.